awx/ui_next/src/screens/User/UserAccess/UserAccessList.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useEffect } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { getQSConfig, parseQueryString } from '@util/qs';
+import { UsersAPI } from '@api';
+import useRequest from '@util/useRequest';
+import PaginatedDataList, {
+  ToolbarAddButton,
+} from '@components/PaginatedDataList';
+import DatalistToolbar from '@components/DataListToolbar';
+import UserAccessListItem from './UserAccessListItem';
+
+const QS_CONFIG = getQSConfig('roles', {
+  page: 1,
+  page_size: 20,
+  order_by: 'id',
+});
+// TODO Figure out how to best conduct a search of this list.
+// Since we only have a role ID in the top level of each role object
+// we can't really search using the normal search parameters.
```

Comment:
I'm not sure what this means - what is the top-level of a role object and what are the normal search params?